packages/api/src/libs/queries.ts

```diff
@@ -0,0 +1,51 @@
+// allow for more complex queries, joins and shared queries between services
+import { Libs } from "..";
+import uma from "@uma/sdk";
+import { calcGcr } from "./utils";
+import assert from "assert";
+import bluebird from "bluebird";
+
+export default (libs: Libs) => {
```

Comment:
Similar question/comment about using the expansive type `Libs` here.

Reply:
on the expansiveness of this variable: its mainly a simplification/convention that I've bundled all mutable/query-able state into this variable. Breaking this apart per service is possible, but I don't see the benefit of this since everything is running in the same process. 

 If and when these services are broken out into their own processes it would probably be necessary to change this signature per process.  Would be happy to discuss this more though if you have some ideas.

Reply:
My main argument against it is that it kinda breaks modularity. if you wanted to test or reuse one of these components, you're required to initialize this libs variable with a lot of state that isn't actually needed by the function.

As a second point, from a readability perspective, it makes it a bit opaque as to what's being used by the class. Rather than being able to quickly glance at the signature, you have to read through the implementation.

My naive alternative is just to list the values that you do need. `{ a, b, c }: {a: TypeA, b: TypeB, c: TypeC}`. Since typescript uses duct-typing, this is completely compatible with the top-level libs type, it just allows you to be more specific about what you need/expect, and typescript will verify that the libs type that's being passed in matches. The downside is a bit of verbosity.

Reply:
yea i could see that, basically have each service define its dependencies, while keeping the top level type the same. will give it a try

Reply:
added something like this to each thing that uses the app state:
![image](https://user-images.githubusercontent.com/4429761/121936913-2181dc80-cd18-11eb-9c2c-6dd0f286dc55.png)

Reply:
Yep, that works! It's not 100% obvious what the types of those variables is, but it is clear which parts the function cares about!